node/src/daemon/dns_inspector/dynamic_store_dns_inspector.rs

```diff
@@ -248,13 +248,21 @@ impl StoreWrapperReal {
 
 #[cfg(test)]
 mod tests {
+    use super::super::core_foundation::array::CFArray;
+    use super::super::core_foundation::propertylist::CFPropertyList;
     use super::*;
+    use crate::daemon::dns_inspector::dynamic_store_dns_inspector::{
+        DynamicStoreDnsInspector, StoreWrapper, PRIMARY_SERVICE, SERVER_ADDRESSES,
+    };
+    use crate::daemon::dns_inspector::DnsInspectionError;
     use core_foundation::boolean::CFBoolean;
     use core_foundation::propertylist::CFPropertyListSubClass;
     use core_foundation::string::CFString;
     use std::cell::RefCell;
-    use std::sync::Arc;
-    use std::sync::Mutex;
+    use std::collections::HashMap;
+    use std::net::IpAddr;
+    use std::str::FromStr;
+    use std::sync::{Arc, Mutex};
```

Comment:
This is really problematic when I don't know how this happened; why these imports started to be needed although there is no change elsewhere in this file which would have answered that. 

Do you see the green lines here? Also look at the header of this smaller window, it belong only to one file, right? The fact that we can see new imports that haven't been obviously brought in because of some seeable need from elsewhere in the file means logically that these imports are very likely redundant. 

It doesn't necessarily mean that these things they are about are not used but it may mean - and I do believe so - that they were named unnecessarily given that there is the line saying "use super::* "
Remember, this line says that everything imported in a higher level (in the superior module) than the module you're in is automatically imported into this module inside the higher. 
Here it means that module "tests" is inside module "dynamic_store_dns_inspector" which is the file itself. If we state this there it very likely means that all the green lines or most of them would be covered by the "super*" but by some unknown reason they got in named explicitly.

I recommend to delete all the green lines and try to compile. The compiler will tell you what imports are missing. Easy.

Reply:
Removed these and the compiler had no issue with it

Reply:
They aren't, believe me. Try without them. Don't let yourself get disorient by seeing weird things. This code counts in for Mac. 

Remove the green lines and all should be fine.

Reply:
Seems like I wasn't exact. Removing all the green lines...true. Except that Mutex and Arc. (Look at it again - you can click on this reference though as "outdated". Arc and Mutex were included into a formatting operation. You can deduce because you see how it looked before - in red - and how it looks afterwards - green and together. The change is obvious, still it implies to be a work of the formater. )

So put back Mutex and Arc...and think of the context. Why we need to do this and how we progressed up to this moment.